pkg/crc/machine/start.go

```diff
@@ -915,3 +940,35 @@ func addPodmanSystemConnections(c *types.ConnectionDetails) error {
 	}
 	return nil
 }
+
+func startMicroshift(ctx context.Context, sshRunner *crcssh.Runner, ocConfig oc.Config, pullSec cluster.PullSecretLoader) error {
+	logging.Infof("Starting Microshift service... [takes around 1min]")
+	content, err := pullSec.Value()
+	if err != nil {
+		return err
+	}
+	if err := sshRunner.CopyDataPrivileged([]byte(content), "/etc/crio/openshift-pull-secret", 0600); err != nil {
+		return err
+	}
+	if _, _, err := sshRunner.RunPrivileged("Starting microshift service", "systemctl", "start", "microshift"); err != nil {
+		return err
+	}
+	if err := sshRunner.CopyFileFromVM("/var/lib/microshift/resources/kubeadmin/kubeconfig", constants.KubeconfigFilePath, 0600); err != nil {
+		return err
+	}
+	if _, _, err := sshRunner.RunPrivileged("Copy kubeconfig file to /opt/kubeconfig", "cp", "/var/lib/microshift/resources/kubeadmin/kubeconfig", "/opt/kubeconfig"); err != nil {
+		return err
+	}
+	if _, _, err := sshRunner.RunPrivileged("Change file permission of /opt/kubeconfig", "chmod", "0644", "/opt/kubeconfig"); err != nil {
+		return err
```

Comment:
I'd use `CopyFile` since you just copied the file locally. It will do both copy to the right place and permission adjustments in one go.

Reply:
I don't understand the comment. `sshRunner.CopyFile(constants.KubeconfigFilePath, "/opt/kubeconfig", 0644)` does not work as expected? The first argument to `CopyFile` is a local file, not a file in the VM, so I don't understand why you mention the `core` user.